rfcs/2025-05-05-client-certificates.md

```diff
@@ -0,0 +1,129 @@
+# Client certificates for high volume clients
+
+## Problem
+
+The absense of user and client identification benefits privacy, but it requires separately authorizing subscription for each messaging queue, that simply doesn't scale even for the current traffic and network size.
+
+While chat relays (aka super-peers) would reduce the number of subscriptions required for the usual clients, by replacing connections with each group member to 1-3 connections with chat relays per group/community, it would shift the burden to the chat relays, that are also clients.
+
+While self-hosted chat relays may want to retain privacy, this is not needed (and counter-productive) for the chat relays provided by network operators.
+
+Even today, directory service subscribing to all queues may take 15-20 minutes, which is experienced as downtime by the end users.
+
+Notification servers are also, effectively, clients that subscribe to the messaging servers, and it also takes 15-20 minutes.
```

Comment:
it is planned to change notifications protocol from subscription to push though, so it would stop to be a part of a problem

Reply:
Yes, but it's a more complex change, and we have 10s of millions of existing subscriptions to migrate anyway.

It is also necessary for directory and for chat relays.

Reply:
and btw I don't think it will be pushed. Subscription creation needs to change to push, but actual subscription is better to continue using this flow when ntf server connects and presents its certificate.

something to think about.